packages/dataviews/src/dataviews.tsx

```diff
@@ -73,25 +70,23 @@ export default function DataViews< Item >( {
 	paginationInfo,
 	defaultLayouts,
 	selection: selectionProperty,
-	setSelection: setSelectionProperty,
-	onChangeSelection = defaultOnChangeSelection,
+	onChangeSelection,
 }: DataViewsProps< Item > ) {
 	const [ selectionState, setSelectionState ] = useState< string[] >( [] );
 	const isUncontrolled =
-		selectionProperty === undefined || setSelectionProperty === undefined;
+		selectionProperty === undefined || onChangeSelection === undefined;
 	const selection = isUncontrolled ? selectionState : selectionProperty;
-	const setSelection = isUncontrolled
-		? setSelectionState
-		: setSelectionProperty;
 	const [ openedFilter, setOpenedFilter ] = useState< string | null >( null );
 
 	function setSelectionWithChange( value: SelectionOrUpdater ) {
 		const newValue =
 			typeof value === 'function' ? value( selection ) : value;
-		onChangeSelection(
-			data.filter( ( item ) => newValue.includes( getItemId( item ) ) )
-		);
```

Comment:
Aren't we now passing the items? Previously we very extracting the ids and now we don't?

Reply:
No, it's the opposite, we're now passing "strings" while we previously were passing "items".

The extra logic here is to filter items that are in the selection but not in the data. The problem though is that this filtering doesn't really make sense when calling `onChange` since we're calling it from the "inside" meaning we know the ids are in the data already.

Reply:
Oh right, I actually changed it internally to IDs before, I forgot :D